treeherder/model/models.py

```diff
@@ -558,6 +571,9 @@ class Job(models.Model):
     machine = models.ForeignKey(Machine, on_delete=models.CASCADE)
     option_collection_hash = models.CharField(max_length=64)
     job_type = models.ForeignKey(JobType, on_delete=models.CASCADE, related_name='jobs')
+    job_desc = models.ForeignKey(
```

Comment:
The job description information is only needed when a job gets selected. The information about the job type is also used in that view.
The job description should be a column of the `job_type` table. This means we only store the data for a few thousand rows and not for many millions.

Reply:
There is the scenario of the description getting updated in one tree or e.g. try.

Recommendation:
1. If there is no job description for the task, store it independent of tree.
2. If there is a job description and the task is central or autoland, update it.
3. No action for other cases.

This could also be tracked by a `job_type.id` + `repository.id` key.